src/bitcoin/wallet.rs

```diff
@@ -1,18 +1,21 @@
+use std::{cell::RefCell, rc::Rc};
+
 use bdk_wallet::{SignOptions, Wallet as BdkWallet};
 use js_sys::Date;
-use serde_wasm_bindgen::to_value;
-use wasm_bindgen::{prelude::wasm_bindgen, JsError, JsValue};
+use wasm_bindgen::{prelude::wasm_bindgen, JsError};
 
 use crate::{
     result::JsResult,
     types::{
-        Address, AddressInfo, Balance, ChangeSet, CheckPoint, FeeRate, FullScanRequest, KeychainKind, Network, Psbt,
-        Recipient, SyncRequest, Update,
+        AddressInfo, Balance, ChangeSet, CheckPoint, FullScanRequest, KeychainKind, LocalOutput, Network, Psbt,
+        SyncRequest, Update,
     },
 };
 
+use super::TxBuilder;
+
 #[wasm_bindgen]
```

Comment:
As discussed internally (feel free to re-phrase):

```suggestion
// We wrap a `BdkWallet` in `Rc<RefCell<...>>` because `wasm_bindgen` do not
// support Rust's lifetimes. This allows us to forward a reference to the
// internal wallet when using `build_tx` and to enforce the lifetime at runtime
// and to preserve "safe mutability".
#[wasm_bindgen]
```